_docs/services/app-autoscaler.md

```diff
@@ -0,0 +1,472 @@
+| instance_min_count                   | int                        | true    | minimal number of instance count for this schedule                         |
+| instance_max_count                   | int                        | true    | maximal number of instance count for this schedule                         |
+| initial_min_instance_count           | int                        | false   | the initial minimal number of instance count for this schedule             |
+
+
+
+The above JSON key/value pair information is sourced from: https://github.com/cloudfoundry/app-autoscaler-release/blob/main/docs/policy.md#schedules
+
+### Specific Schedule Example
+
+The Scaling Policy below has been manipulated to be artificially low to result in the application scaling to 4 application instances during a 30 minute window:
+
+```JSON
+{
+  "instance_min_count": 1,
+  "instance_max_count": 4,
+  "scaling_rules":
+  [
+   {
+    "metric_type": "memoryused",
+    "breach_duration_secs": 60,
+    "threshold": 0,
+    "operator": ">",
+    "cool_down_secs": 60,
+    "adjustment": "+1"
+   }   
+  ],
+  "schedules": {
+    "timezone": "America/New_York",
+    "specific_date": [
+      {
+        "start_date_time": "2024-04-23T16:30",
+        "end_date_time": "2024-04-23T17:00",
+        "instance_min_count": 1,
+        "instance_max_count": 4,
+        "initial_min_instance_count": 2
+      }
+    ]
+  }
+}
+```
+
+An additional complex example of combining Scaling Rules, and both types of Schedules (recurring and specific) can be found at [https://github.com/cloudfoundry/app-autoscaler-release/blob/main/docs/assets/fullpolicy.json](https://github.com/cloudfoundry/app-autoscaler-release/blob/main/docs/assets/fullpolicy.json).
+
+### Scheduling Constraint
+
+If one schedule overlaps another, the one which **starts** first will be guaranteed, while the later one is completely ignored. For example:
+
+    - Schedule #1:  --------sssssssssss----------------------------
+    - Schedule #2:  ---------------ssssssssssssss-----------------
+    - Schedule #3:  --------------------------sssssssss------------
+
+    With the above definition, schedule #1 and #3 will be applied, while schedule #2 is ignored.
+
+## Custom Metrics
+
+### What are custom metrics?
+
+You can define your own metric name and emit that metric to App-AutoScaler to trigger dynamic scaling. 
+
+To scale with a custom metric, the application needs to emit its own metric to App-AutoScaler's MetricsForwarder URL. Given the metric submission is proceeded inside an application, an App-AutoScaler specific service instance credential is required to authorize the access.  
+
+
+
+{% mermaid_diagram id-prefix="Figure-2.-custom-metrics-data-flow" %}
+graph LR
+    subgraph App-AutoScaler
+        E[MetricsForwarder]
+    end
+
+    subgraph CF
+        C[My App] -->|POST to /v1/apps/app_guid/metrics| E[MetricsForwarder]
```

Comment:
So to create custom metrics you need to POST to `/v1/apps/app_guid/metrics`? I wasn't totally sure

Reply:
Yes, the app on CF needs to send a POST to the App-AutoScaler API metrics endpoint which is MetricsForwarder

MetricsForward then injects the metric into log-cache and EventGenerator completes the loop and pulls it off of log-cache to it can be acted on